internal/controller/ansibleRun/ansibleRun.go

```diff
@@ -278,3 +330,69 @@ func (c *external) Delete(ctx context.Context, mg resource.Managed) error {
 
 	return nil
 }
+
+func getLastApplied(observed *v1alpha1.AnsibleRun) (*v1alpha1.AnsibleRun, error) {
+	lastApplied, ok := observed.GetAnnotations()[v1.LastAppliedConfigAnnotation]
+	if !ok {
+		return nil, nil
+	}
+
+	last := &v1alpha1.AnsibleRun{}
+	if err := json.Unmarshal([]byte(lastApplied), last); err != nil {
+		return nil, errors.Wrap(err, errUnmarshalTemplate)
+	}
+
+	if last.GetName() == "" {
+		last.SetName(observed.GetName())
+	}
+
+	return last, nil
+}
+
+func (c *external) handleLastApplied(last, desired *v1alpha1.AnsibleRun) (managed.ExternalObservation, error) {
+	isUpToDate := false
+
+	if last != nil && equality.Semantic.DeepEqual(last, desired) {
+		// Mark as up-to-date since last is equal to desired
+		isUpToDate = true
+	}
+
+	if !isUpToDate {
```

Comment:
When it's not up-to-date, how we trigger the Ansible roles/playbooks? I couldn't find it, or maybe I missed something.

Reply:
You are totally  right, I forget to call/implement the cmdFunc here, I will fix this 👍

Reply:
fixed